mesonbuild/dependencies/python.py

```diff
@@ -347,6 +284,68 @@ def find_libpy_windows(self, env: 'Environment', limited_api: bool = False) -> N
         self.link_args = largs
         self.is_found = True
 
+class PythonPkgConfigDependency(PkgConfigDependency, _PythonDependencyBase):
```

Comment:
I may be missing something, but the only thing this patch does is to move the code around unmodified in the same file. What's the point? Why does the code need to be moved?

Reply:
The change to the Windows checking logic introduced in 663700c means that a `PythonPkgConfigDependency` might now be found where previously only a `PythonSystemDependency` was found.

The `find_libpy_windows` function previously only existed on `PythonSystemDependency`.

Moving this function (and all the functions it calls transitively) into the base class seemed like the easiest solution to the AttributeErrors that I was seeing during the initial development of this patch. It is a bit brutal though, I admit.

The change in 663700c is not enough to resolve the initial issue that this PR addresses. Ultimately the change in b26f306 is necessary in order to find the right library under mingw.

Reply:
The diff is deceptive, since a giant block of code gets moved around but it *looks* like a bunch of classes got moved around some functions, but what actually happened is some functions got moved from one class to another.

It's not easy to tell that the moved code starts halfway through a class definition (thus indicating that the semantics changed).